samtranslator/model/eventsources/push.py

```diff
@@ -253,6 +258,31 @@ def _depend_on_lambda_permissions(self, bucket, permission):
 
         return bucket
 
+    def _depend_on_lambda_permissions_using_tag(self, bucket, permission):
+        """
+        Since conditional DependsOn is not supported this undocumented way of 
```

Comment:
Add the explanation for this workaround (you can link to the StackOverflow but please also give an inline summary of how this works)